### ISSUE TITLE
[a11y]: Make the tooltip position configurable in DismissibleTag

### ISSUE THREAD

**Issue by kubijo** [carbon-design-system/carbon#19891]
### Package

@carbon/react

### Package version

1.86.0

### React version

19.1.0

### Automated testing tool and ruleset

None, bumped into it during development

### Assistive technology

no

### Description

You are forcing the tooltip to be positioned at the bottom in [`DismissibleTag.tsx:190`](https://github.com/carbon-design-system/carbon/blob/main/packages/react/src/components/Tag/DismissibleTag.tsx#L190) and it makes for a bad time when overflow comes into play.

<img width="300" height="133" alt="Image" src="https://github.com/user-attachments/assets/95a17554-26de-4de0-afa6-84d03c4678b5" />

### WCAG 2.1 Violation

Dunno, but the tooltip text is not visible fully

### Reproduction/example

Not much to do here, the description says it all.

### Steps to reproduce

Not much to do here, the description says it all.

### Suggested Severity

Severity 3 = User can complete task, and/or has a workaround within the user experience of a given component.

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [x] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems
